CondFormats/GEMObjects/src/GEMeMap.cc

```diff
@@ -0,0 +1,169 @@
+#include "CondFormats/GEMObjects/interface/GEMeMap.h"
+#include "CondFormats/GEMObjects/interface/GEMROMapping.h"
+#include "DataFormats/MuonDetId/interface/GEMDetId.h"
+#include "DataFormats/FEDRawData/interface/FEDNumbering.h"
+
+GEMeMap::GEMeMap():
+  theVersion("") {}
+
+GEMeMap::GEMeMap(const std::string & version):
+  theVersion(version) {}
+
+GEMeMap::~GEMeMap() {}
+
+const std::string & GEMeMap::version() const {
+  return theVersion;
+}
+
+void GEMeMap::convert(GEMROMapping & romap) {
+  // fed->amc->geb mapping to GEMDetId
+  for (auto imap : theChamberMap_) {
+    for (unsigned int ix=0;ix<imap.fedId.size();ix++) {
+      GEMROMapping::chamEC ec;
+      ec.fedId = imap.fedId[ix];
+      ec.amcNum = imap.amcNum[ix];
+      ec.gebId = imap.gebId[ix];
+
+      GEMROMapping::chamDC dc;
+      dc.detId = GEMDetId((imap.gemNum[ix] > 0) ? 1:-1, 1, abs(imap.gemNum[ix]/1000), abs(imap.gemNum[ix]/100%10), abs(imap.gemNum[ix]%100), 0);
+      dc.vfatVer = imap.vfatVer[ix];
+
+      // std::cout << " fedId " << ec.fedId
+      //           << " amcNum " << ec.amcNum
+      //           << " gebId " << ec.gebId
+      //           << " vfatVer " << dc.vfatVer
+      //           << " id " << dc.detId
+      //           << std::endl;
+      romap.add(ec, dc);
+    }
+  }
+  // chamberType to vfatType
+  for (auto imap : theVFatMap_) {
+    for (unsigned int ix=0;ix<imap.vfatAdd.size();ix++) {
+      GEMDetId gemId((imap.gemNum[ix] > 0) ? 1:-1, 1, abs(imap.gemNum[ix]/1000), abs(imap.gemNum[ix]/100%10), abs(imap.gemNum[ix]%100), imap.iEta[ix]);
+      
+      GEMROMapping::vfatEC ec;
+      ec.detId = gemId.chamberId();
+      ec.vfatAdd = imap.vfatAdd[ix] & chipIdMask_;
+      
+      GEMROMapping::vfatDC dc;
+      dc.vfatType = imap.vfatType[ix];
+      dc.detId = gemId;
+      dc.localPhi = imap.localPhi[ix];
+
+      std::cout << " vfatAdd " << ec.vfatAdd
```

Comment:
@jshlee @ggovi is this cout needed? This is not thread safe in case

Reply:
@jshlee please remove the cout if not required, or replace them with the edm::Log functions

Reply:
removed